pkg/pvc/pvc_protected_entity.go

```diff
@@ -201,54 +200,72 @@ func (this PVCProtectedEntity) DeleteSnapshot(ctx context.Context, snapshotToDel
 			return false, errors.Wrapf(err, "Could not retrieve snapshot configmap %s for %s", snapConfigMapName,
 				this.id.String())
 		}
+		this.logger.Infof("PVCProtectedEntity: No snapshot config map avaiable with name: %s", snapConfigMapName)
 		// No configmap so no snapshots
 		return false, nil
 	}
 
-	_, snapPVCInfoExists := snapConfigMap.BinaryData[snapshotToDelete.String()]
-	if snapPVCInfoExists {
-		delete(snapConfigMap.BinaryData, snapshotToDelete.String())
-		delete(snapConfigMap.BinaryData, "peinfo-"+snapshotToDelete.String())
-		updatedSnapConfigMap, err := this.ppetm.clientSet.CoreV1().ConfigMaps(pvc.Namespace).Update(snapConfigMap)
-		if err != nil {
-			return false, errors.Wrapf(err, "Could not update snapshot configmap %s for %s", snapConfigMapName,
-				this.id.String())
-		}
-		if updatedSnapConfigMap.BinaryData == nil {
-			// no snapshot after the update. So, clean up the config map
-			var zeroSecondsGracePeriod int64
-			zeroSecondsGracePeriod = 0
-			err := this.ppetm.clientSet.CoreV1().ConfigMaps(pvc.Namespace).Delete(updatedSnapConfigMap.Name, &metav1.DeleteOptions{GracePeriodSeconds: &zeroSecondsGracePeriod})
-			if err != nil {
-				return false, errors.Wrapf(err, "Could not delete snapshot configmap %s for %s", snapConfigMapName,
-					this.id.String())
-			}
-		}
-	}
 	components, err := this.GetComponents(ctx)
 	if err != nil {
 		return false, errors.Wrap(err, "Could not retrieve components")
 	}
 	if len(components) != 1 {
 		return false, errors.New(fmt.Sprintf("Expected 1 component, %s has %d", this.id.String(), len(components)))
 	}
-	subSnapshots, err := components[0].ListSnapshots(ctx)
-	if err != nil {
-		return false, errors.Wrapf(err, "Subcomponent peid %s list snapshot", components[0].GetID())
+	this.logger.Infof("PVCProtectedEntity: Retrieved components for the pvc protected entity id: %s", components[0].GetID().String())
+
+	subSnapshots := make([]astrolabe.ProtectedEntitySnapshotID, 0)
+	paraVirtComponentType := false
+	if components[0].GetID().GetPeType() == "paravirt-pv" {
```

Comment:
Maybe instead of listing the snapshots we should just blindly try to delete the snapshot ID we have.  If it fails, just log it and continue.  That would simplify this section a lot.